cookbooks/fb_collectd/libraries/default.rb

```diff
@@ -0,0 +1,43 @@
+# vim: syntax=ruby:expandtab:shiftwidth=2:softtabstop=2:tabstop=2
+#
+# Copyright (c) 2016-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+#
+
+module FB
+  # collectd utility functions
+  class Collectd
+    # Internal helper function to generate /etc/collectd/collectd.conf entries
+    def self._gen_collectd_conf_entry(k, v, i = 0)
+      indent = ' ' * i
+      if v.is_a?(Array)
+        s = ''
+        v.each do |vv|
+          s += self._gen_collectd_conf_entry(k, vv, i)
```

Comment:
slightly simpler if you add ` + "\n"` to the end of htis line... but actually rather than building strings like this, why not build an array and join them all at the end rather than having to keep track of newlines at all?